File: Hw05/code/montageDigits.m
function avgImage = montageDigits(x)
numImages = size(x, 4);
nx = 20;
ny = ceil(numImages/nx);
montage(x, 'Size', [ny nx]);

if nargout > 0, %compute avgImg
    avgImage = squeeze(mean(x, 4));
end

File: Hw05/code/montageDigits.py
import numpy as np
import matplotlib.pyplot as plt
from skimage.util.montage import montage2d

def montageDigits(x):
    num_images = x.shape[2]
    m = montage2d(x.transpose(2, 0, 1))
    plt.imshow(m, cmap='gray')
    plt.show()

    return np.mean(x, axis=2)
